codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -288,75 +309,93 @@ module Make = (
         | parsedEvents => parsedEvents
         }
 
-        pageUnsafe.items
-        ->Belt.Array.zip(parsedEvents)
-        ->Belt.Array.map(((item, event)): Types.eventBatchQueueItem => {
-          let {block, transaction, log: {logIndex}} = item
+        pageUnsafe.items->Belt.Array.mapWithIndex((index, item): Types.eventBatchQueueItem => {
+          let {block, log} = item
           let chainId = chain->ChainMap.Chain.toChainId
-          let (event, eventMod) = switch event
-          ->getNullableExn(~msg="Event was unexpectedly parsed as undefined", ~logger)
-          ->Converters.convertHyperSyncEvent(
-            ~eventModLookup,
+          let topic0 = log.topics->Js.Array2.unsafe_get(0)
+
+          let eventMod = switch eventModLookup->EventModLookup.get(
+            ~sighash=topic0,
             ~contractAddressMapping,
-            ~log=item.log,
-            ~block,
-            ~transaction,
-            ~chain,
+            ~contractAddress=log.address,
           ) {
-          | Ok(v) => v
-          | Error(exn) =>
-            let logger = Logging.createChildFrom(
-              ~logger,
-              ~params={"chainId": chainId, "blockNumber": block.number, "logIndex": logIndex},
-            )
-            exn->ErrorHandling.mkLogAndRaise(~msg="Failed to convert decoded event", ~logger)
-          }
-          {
-            timestamp: block.timestamp,
-            chain,
-            blockNumber: block.number,
-            logIndex,
-            event,
-            eventMod,
+          | None => {
+              let logger = Logging.createChildFrom(
+                ~logger,
+                ~params={
+                  "chainId": chainId,
```

Comment:
I think we should explore our own "context" modelling soon. For instance I think that chainId probably already exists on the parent logger but have no idea and deduplication isn't handled.

Reply:
Yes, and make it lazily computed (so the object is allocated only when an error is thrown)